src/components/TodoModal/TodoModal.tsx

```diff
@@ -3,10 +1,30 @@
+import React, { useEffect, useState } from 'react';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { Loader } from '../Loader/Loader';
+import { getUser } from '../../api';
+
+interface Props {
+  todo: Todo;
+  closeModal: () => void;
+}
+
+export const TodoModal: React.FC<Props> = ({ todo, closeModal }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoadingUser, setIsLoadingUser] = useState(true);
+
+  useEffect(() => {
+    setIsLoadingUser(true);
+    getUser(todo.userId)
+      .then(userData => setUser(userData))
+      .finally(() => setIsLoadingUser(false));
+  }, [todo.userId]);
```

Comment:
You are directly interacting with the API inside the component. This could lead to performance issues and make the component less reusable. Consider moving the API calls outside of the component, for instance using custom hooks. This way, you could separate the data-fetching logic from the component, making your code cleaner and easier to test.